include/dos_system.h

```diff
@@ -305,10 +314,34 @@ class DOS_Drive {
 	virtual bool isRemovable(void)=0;
 	virtual Bits UnMount(void)=0;
 
-	const char * GetInfo() const { return info; }
+	DosDriveType GetType() const
+	{
+		return type;
+	}
+	const char *GetInfo() const
+	{
+		return info;
+	}
+	std::string GetInfoString() const
+	{
+		switch (type) {
+		case DosDriveType::Local:
+			return MSG_Get("MOUNT_TYPE_LOCAL_DIRECTORY") +
+			       std::string(" ") + info;
+		case DosDriveType::Cdrom:
+			return MSG_Get("MOUNT_TYPE_CDROM") + std::string(" ") + info;
+		case DosDriveType::Fat:
+			return MSG_Get("MOUNT_TYPE_FAT") + std::string(" ") + info;
+		case DosDriveType::Iso:
+			return MSG_Get("MOUNT_TYPE_ISO") + std::string(" ") + info;
+		case DosDriveType::Virtual: return MSG_Get("MOUNT_TYPE_VIRTUAL");
+		default: return MSG_Get("MOUNT_TYPE_UNKOWN");
+		}
+	}
 
 	char curdir[DOS_PATHLENGTH];
 	char info[256];
+	DosDriveType type;
```

Comment:
Can use default initializers for these members:
```c++
	char curdir[DOS_PATHLENGTH] = {};
	char info[256] = {};
	DosDriveType type = DosDriveType::Unknown;
```

Which should fix the effective C++ warning.

Reply:
Done accordingly. Thanks!